chia/simulator/setup_nodes.py

```diff
@@ -409,103 +358,99 @@ async def setup_full_system_inner(
-            False,
-            10,
-            True,
-            connect_to_daemon=connect_to_daemon,
-            db_version=db_version,
-        )
-        for i in range(2)
-    ]
-    nodes = [await fni.__anext__() for fni in full_node_iters]
-    node_apis = [fni._api for fni in nodes]
-    full_node_0_port = node_apis[0].full_node.server.get_port()
-    farmer_iter = setup_farmer(
-        shared_b_tools,
-        shared_b_tools.root_path / "harvester",
-        shared_b_tools.config["self_hostname"],
-        consensus_constants,
-        full_node_0_port,
-    )
-    farmer_service = await farmer_iter.__anext__()
-    harvester_iter = setup_harvester(
-        shared_b_tools,
-        shared_b_tools.root_path / "harvester",
-        UnresolvedPeerInfo(shared_b_tools.config["self_hostname"], farmer_service._server.get_port()),
-        consensus_constants,
-    )
-    vdf1_port = uint16(find_available_listen_port("vdf1"))
-    vdf2_port = uint16(find_available_listen_port("vdf2"))
-
-    timelord_iter = setup_timelord(
-        full_node_0_port, False, consensus_constants, b_tools.config, b_tools.root_path, vdf_port=vdf1_port
-    )
-    timelord_bluebox_iter = setup_timelord(
-        uint16(1000), True, consensus_constants, b_tools_1.config, b_tools_1.root_path, vdf_port=vdf2_port
-    )
-    harvester_service = await harvester_iter.__anext__()
-    harvester = harvester_service._node
-
-    async def num_connections() -> int:
-        count = len(harvester.server.all_connections.items())
-        return count
-
-    await time_out_assert_custom_interval(10, 3, num_connections, 1)
-    node_iters = [
-        introducer_iter,
-        harvester_iter,
-        farmer_iter,
-        setup_vdf_clients(shared_b_tools, shared_b_tools.config["self_hostname"], vdf1_port),
-        timelord_iter,
-        full_node_iters[0],
-        full_node_iters[1],
-        setup_vdf_client(shared_b_tools, shared_b_tools.config["self_hostname"], vdf2_port),
-        timelord_bluebox_iter,
-    ]
-    if connect_to_daemon:
-        node_iters.append(daemon_iter)
-    timelord = await timelord_iter.__anext__()
-    vdf_clients = await node_iters[3].__anext__()
-    timelord_bluebox_service = await timelord_bluebox_iter.__anext__()
-    timelord_bluebox = timelord_bluebox_service._api
-    timelord_bluebox_server = timelord_bluebox_service._node.server
-    vdf_bluebox_clients = await node_iters[7].__anext__()
-    ret = (
-        node_apis[0],
-        node_apis[1],
-        harvester,
-        farmer_service._node,
-        introducer,
-        timelord,
-        vdf_clients,
-        vdf_bluebox_clients,
-        timelord_bluebox,
-        timelord_bluebox_server,
-    )
-    return daemon_ws, node_iters, ret
+    async with setup_introducer(shared_b_tools, uint16(0)) as introducer_service:
+        introducer = introducer_service._api
+        introducer_server = introducer_service._node.server
+        async with AsyncExitStack() as astack:
```

Comment:
I'd say that `exit_stack` would be more clear.  The exit'ness of the stack seems more descriptive than the async'ness.  As in, it isn't an async call stack, for example.

Reply:
This is a slight modification of the way they do it in the docs here: https://docs.python.org/3/library/contextlib.html#contextlib.AsyncExitStack (I only added the `a` because of a conflict in some section of the code with a non-async exit stack)

Reply:
![image](https://github.com/Chia-Network/chia-blockchain/assets/543719/f0f57fd9-9100-403c-a752-d682f9347397)

Our existing uses.  `:]`

But...  I didn't actually raise a concern so...  I just prefer non-abbreviations.  And sure, sometimes you do want to clarify async vs. not and maybe whatever the pattern is should be consistent instead of conditional on whether they are used in the same scope or not.

Reply:
Ah I didn't know we were using this at all, so yeah I'd rather be consistent across our own repo

Reply:
Renamed to async_exit_stack